Microsoft.Toolkit.Uwp.UI/Extensions/Mouse/Mouse.cs

```diff
@@ -0,0 +1,114 @@
+      /// <param name="value">Target cursor type value.</param>
+      public static void SetCursor(FrameworkElement element, CoreCursorType value)
+      {
+         element.SetValue(CursorProperty, value);
+      }
+
+      /// <summary>
+      /// Get the current <see cref="CoreCursorType"/>.
+      /// </summary>
+      /// <param name="element">Object where the selector cursor type should be shown.</param>
+      /// <returns>Cursor type set on target element.</returns>
+      public static CoreCursorType GetCursor(FrameworkElement element)
+      {
+         return (CoreCursorType)element.GetValue(CursorProperty);
+      }
+
+      private static void CursorChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+      {
+         FrameworkElement element = d as FrameworkElement;
+         if (element == null)
+         {
+            throw new NullReferenceException(nameof(element));
+         }
+
+         CoreCursorType value = (CoreCursorType)e.NewValue;
+
+         lock (CursorLock)
+         {
+            if (!Cursors.ContainsKey(value))
+            {
+               Cursors[value] = new CoreCursor(value, 1);
+            }
+
+            // make sure event handlers are not attached twice to element
+            element.PointerEntered -= Element_PointerEntered;
+            element.PointerEntered += Element_PointerEntered;
+            element.PointerExited -= Element_PointerExited;
+            element.PointerExited += Element_PointerExited;
+            element.Unloaded -= ElementOnUnloaded;
+            element.Unloaded += ElementOnUnloaded;
+         }
+      }
+
+      private static void Element_PointerEntered(object sender, PointerRoutedEventArgs e)
+      {
+         CoreCursorType cursor = GetCursor((FrameworkElement)sender);
+         Window.Current.CoreWindow.PointerCursor = Cursors[cursor];
+      }
+
+      private static void Element_PointerExited(object sender, PointerRoutedEventArgs e)
+      {
+         // when exiting change the cursor to the target Mouse.Cursor value of the new element
+         CoreCursor cursor;
+         if (e.OriginalSource is FrameworkElement newElement)
+         {
+            cursor = Cursors[GetCursor(newElement)];
+         }
+         else
+         {
+            cursor = DefaultCursor;
+         }
+
+         Window.Current.CoreWindow.PointerCursor = cursor;
+      }
+
+      private static void ElementOnUnloaded(object sender, RoutedEventArgs routedEventArgs)
+      {
+         // when the element is programatically unloaded, reset the cursor back to default
+         // this is necessary when click triggers immediate change in layout and PointerExited is not called
+         Window.Current.CoreWindow.PointerCursor = DefaultCursor;
+      }
+
+      private static readonly object CursorLock = new object();
+      private static readonly CoreCursor DefaultCursor = new CoreCursor(CoreCursorType.Arrow, 1);
```

Comment:
Couldn't we grab the initial Window.Current.CoreWindow.PointerCursor instead of making our own cursor? (e.g. in the CreateDefaultValueCallback in the RegisterAttached)